module/zfs/zfs_replay.c

```diff
@@ -337,8 +337,10 @@ zfs_replay_create_acl(void *arg1, void *arg2, boolean_t byteswap)
 	xva.xva_vattr.va_nblocks = lr->lr_gen;
 	xva.xva_vattr.va_fsid = dnodesize;
 
-	error = dmu_object_info(zfsvfs->z_os, lr->lr_foid, NULL);
-	if (error != ENOENT)
+	error = dnode_hold_impl(zfsvfs->z_os, objid,
```

Comment:
I know I suggested you drop the helper function before, but perhaps it would be nice to have a `dnode_try_claim()` wrapper to make it very clear what this check if for.